libvast/src/system/query_supervisor.cpp

```diff
@@ -53,7 +53,12 @@ query_supervisor_actor::behavior_type query_supervisor(
         const index_client_actor& client) {
       VAST_DEBUG(self, "got a new query for", qm.size(),
                  "partitions:", get_ids(qm));
-      VAST_ASSERT(!qm.empty());
+      if (qm.empty()) {
+        // Nothing to do.
+        self->send(client, atom::done_v);
+        self->send(master, atom::worker_v, self);
+        return;
+      }
```

Comment:
👍 and add an assertion here whether `!qm.empty()` with a comment why.

Reply:
I'm not sure this is better; this makes the logic in the index much more complicated. (since we have to return `atom::done`, return the worker, and maybe erase the lookup handle if these were the last partitions)

Reply:
Can you at least add a comment with a TODO for this question?

Reply:
Sure, added a TODO-comment.